src/rp/util/string.clj

```diff
@@ -28,3 +28,8 @@
   (some-> s
           parse-double*
           double))
+
+(defn non-blank-string?
+  "Returns true if x is a non-blank string"
+  [x] (and (string? x)
+           (not (str/blank? x))))
```

Comment:
What do folks think about, as a pattern, relying less on traditional predicates that return literally `true` or `false` and instead leveraging the truthy patterns that we find throughout Clojure? In that case, this would look something like:

```clj
(defn non-blank-string
  "Returns x if it is a non-blank string, falsey otherwise."
  [x]
  (when (and (string? x)
             (not (str/blank? x)))
    x)
```

I think this would fit nicely with `if-let` and `when-let` patterns as well as for things like spec conformation, amongst other things.